### DISCUSSION TITLE
Image location

### DISCUSSION THREAD

**Discussion by ceford** [max123kl/J4.docs#16] [General]
In Mediwiki everything appears to be at the root level, even images. However, when rendered you find that images are in subfolders like this:

/images/thumb/d/d3/J4.x-introduction-to-joomla-cassiopeia-en.png/800px-J4.x-introduction-to-joomla-cassiopeia-en.png

I created a tutorial for Docusaurus but there were no guidlines on where to put images. So I put them in the same folder as the md file. We need to think about that. Should we even be thinking about a separate image server? 

This is what a rendered Mediawiki image looks like:

```
<img alt="J4.x-introduction-to-joomla-cassiopeia-en.png" 
src="/images/thumb/d/d3/J4.x-introduction-to-joomla-cassiopeia-en.png/800px-J4.x-introduction-to-joomla-cassiopeia-en.png" 
decoding="async" class="thumbborder" 
srcset="/images/d/d3/J4.x-introduction-to-joomla-cassiopeia-en.png 1.5x" data-file-width="1000" 
data-file-height="508" width="800" height="406">
```

**Comment by carlitorweb** [comment]
I like when we can put the images associate with the doc, in the same folder of the doc. This prevents contributors from creating wrong-defined structures in a separate image folder. 
So we have the .md tutorial/doc and there, a folder called images with all the images used by the .md file

**Reply by max123kl** [reply]
We have to define how images appearing on multiple pages should be handled

**Reply by carlitorweb** [reply]
Migrating the content, we should kown what image appear in other place, but in a new brand doc, how we wshould kown if the images used there will be used in other future doc?

**Reply by max123kl** [reply]
Thats why I prefer to have an images folder and create symlinks to the text folders

**Reply by carlitorweb** [reply]
but then we need create a structure too for that image folder, and mean we need try future contribution follow it and not create wrong folders or use wrong folders for they images.

**Reply by max123kl** [reply]
Naming of the files will be most important
Best will be if we can put the selection of the right storage folder into a script, that follows a PR merge

**Comment by alann60** [comment]
With static site generators I have looked at it seems common practice for every page to reside in its own folder with images in the same folder.

Storage capacity will be, I would think, a major consideration.

While it may exist, I've not seen anywhere that you can check what images there are so that they can be utilised in multiple documents.

That said, document writers use different ways to emphasise buttons etc in images, some use circles, some use squares, some use arrows, some use highlighting - a consistent approach through guidance would be beneficial. Again, maybe there is such guidance already?

If this could be done moving forwards it will be beneficial. As not everyone has software or skills for preparing or editing images, consistent image marking makes it feasible to swap one out without worrying about matching a replacement with others that are OK on the page

**Reply by max123kl** [reply]
It doesn't really matter where an image is finally dropped on the storage. The CMS (Wiki), whichever it is, only needs to know where. With so-called flat-file CMSs such as Hugo or Grav, there is often no database for the organization in the background. This makes it difficult to create a link to an image across several folder hierarchies without problems. With Joomla, you have the images folder and can create your own structure there with subdirectories. This has the advantage that you can find all the images in one central place, and there are also no unnecessary duplicates of the image files if an image is used several times. The database determines the internal link for you as soon as you place it on a page.

A [list of media files](https://docs.joomla.org/Special:ListFiles) exists. However, the search for a specific file is only successful if you know the name as precisely as possible. Unfortunately, hardly any authors know the guidelines for naming image files, and so it is a stroke of luck if the file you are looking for is found. Since the database manages everything in the background, hardly anyone notices that the system is not working optimally,

The guidelines are like the naming of the image files, they exist but hardly anyone knows or finds them. One of our priority tasks will be to establish and enforce a workflow.

I agree with you about the image processing skills - the lack of software is certainly not a reason. People who work so closely with Joomla are all computer-savvy to a certain extent. They have no problem acquiring software.

**Reply by ceford** [reply]
My advice: avoid images as much as possible. They are not translatable so translators either have to capture the same image in their own language or use th English image. For that reason I removed all of the images of Toolbars in J4 Help Screens on the grounds that they are in the main screenshot anyway. And inline images of buttons or drop-down lists with English labels are bad too.

In the site I designed that is now over 10 years old we get several thousand new photographs each year. Each one is named with the number of the record it resfers to so xxxxxx-n where xxxxxx is the record and n the sequence. That would not be recommended now as images are supposed to seo too. Where we put the images to start (MediaWiki or GitHub or somewhere else)  and where they end up for delivery are separate problems. But, as you will see from my uploaded docs, having them in Mediwiki solves the duplication problem for the time being. 

This is clearly something to ponder and will depend on deployment.

**Reply by alann60** [reply]
> My advice: avoid images as much as possible.

That's an interesting view. I've always been of the opinion that a picture speaks a thousand words. But when I think about it, if one unnecessary image results in half a dozen or more translated ones, over the life of a project that is a significant amount of work and storage.

Definitely something to consider.
